src/main/java/nl/ou/debm/producer/CGenerator.java

```diff
@@ -522,11 +542,15 @@ public Function getFunction(int currentDepth, DataType type, Boolean withParamet
             var wantedType = type;
             // if no particular return type is wanted, pick a s random one
             if (wantedType == null) {
-                var keys = callableFunctionsByReturnType.keySet().stream().toList();
+                var keys = mCallableFunctionsByReturnType.keySet().stream().toList();
                 wantedType = keys.get(ThreadLocalRandom.current().nextInt(0, keys.size()));
             }
             // pick a random function on the basis of the return type
-            return callableFunctionsByReturnType.get(wantedType).get(ThreadLocalRandom.current().nextInt(0, callableFunctionsByReturnType.get(wantedType).size()));
+            var r =  mCallableFunctionsByReturnType.get(wantedType).get(ThreadLocalRandom.current().nextInt(0, mCallableFunctionsByReturnType.get(wantedType).size()));
+            if(r.getName().startsWith("function_")){
+                System.out.println("ERROR");
+            }
```

Comment:
This looks like a debug hook. Should it be removed? Or turned into an assert/throw RuntimeException?